testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/SocketsAndInterfacesTestCase.java

```diff
@@ -129,9 +154,7 @@ public void testAddUpdateRemove() throws Exception {
 
         logger.info("Restarting server.");
 
-        // reload server
-        op = createOpNode(null, "reload");
-        result = executeOperation(op);
+        ServerReload.executeReloadAndWaitForCompletion(getModelControllerClient());
```

Comment:
@bstansberry yes but if you call add for socket-binding with allow-resource-service-restart header it will be added on execution of that operation.
We do have few tests in our testsuite that verify this, for example ListenerTestCase adds/removes (new) socket bindings & listeners all without need for reload.